lib/view_model/active_record/update_operation.rb

```diff
@@ -356,6 +356,28 @@ def build_update_for_single_association(association_data, association_update_dat
       end
     end
 
+    def functional_update_contents(update, update_context)
+      refs = update
+               .update_datas(update_context.referenced_update_data)
```

Comment:
This one is a bit awkward. Note that we want to use the update datas inside `UpdateContext#build` for preloading, which happens before `build!`. `resolve_reference`, which returns update ops, which are constructed in build, cannot work before `build!` is complete. We could add a parallel `resolve_reference_update_data` (± naming), that works before `build!` and gives us similar safety, but I don't think we can use the existing resolution method.